File: Forge/src/main/java/hantonik/atomic/core/inventory/menu/AtomicContainerMenu.java
package hantonik.atomic.core.inventory.menu;

import lombok.Getter;
import net.minecraft.core.BlockPos;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.inventory.AbstractContainerMenu;
import net.minecraft.world.inventory.MenuType;
import net.minecraft.world.item.ItemStack;

public class AtomicContainerMenu extends AbstractContainerMenu {
    @Getter
    private final BlockPos blockPos;

    protected AtomicContainerMenu(MenuType<?> type, int id, BlockPos pos) {
        super(type, id);

        this.blockPos = pos;
    }

    @Override
    public ItemStack quickMoveStack(Player p_38941_, int p_38942_) {
        return ItemStack.EMPTY;
    }

    @Override
    public boolean stillValid(Player player) {
        return player.distanceToSqr(this.blockPos.getX() + 0.5, this.blockPos.getY() + 0.5, this.blockPos.getZ() + 0.5) <= 64;
    }
}

File: Forge/src/main/kotlin/hantonik/atomic/core/inventory/menu/AtomicContainerMenu.kt
package hantonik.atomic.core.inventory.menu

import net.minecraft.core.BlockPos
import net.minecraft.world.entity.player.Player
import net.minecraft.world.inventory.AbstractContainerMenu
import net.minecraft.world.inventory.MenuType
import net.minecraft.world.item.ItemStack

open class AtomicContainerMenu protected constructor(type: MenuType<out AtomicContainerMenu>, id: Int, val blockPos: BlockPos) : AbstractContainerMenu(type, id) {
    override fun quickMoveStack(player: Player, index: Int): ItemStack = ItemStack.EMPTY

    override fun stillValid(player: Player) = player.distanceToSqr(this.blockPos.x + 0.5, this.blockPos.y + 0.5, this.blockPos.z + 0.5) <= 64
}